packages/xstate-react/src/useMachine.ts

```diff
@@ -29,6 +30,10 @@ export interface UseMachineOptions<TContext, TEvent extends EventObject> {
    * start at this state instead of its `initialState`.
    */
   state?: StateConfig<TContext, TEvent>;
+  /**
+   * Convenient syntax to create a subscription to the interpreter at the creation time.
+   */
+  subscribe?: (state: AnyState) => void;
```

Comment:
This type is definitely too loose - if we gonna decide to move with this feature it should provide a stricter type to the one who has provided the callback.

Reply:
How do you tighten the types here? Reuse the types of `interpreter.subscribe` argument? Because I tried that, but that subscribe method doesn't infer types.